pkg/broker/ingress/multi_topic_decouple_sink.go

```diff
@@ -53,31 +63,84 @@ type multiTopicDecoupleSink struct {
 
 // Send sends incoming event to its corresponding pubsub topic based on which broker it belongs to.
 func (m *multiTopicDecoupleSink) Send(ctx context.Context, ns, broker string, event cev2.Event) protocol.Result {
-	topic, err := m.getTopicForBroker(ns, broker)
+	topic, err := m.getTopicForBroker(types.NamespacedName{Namespace: ns, Name: broker})
 	if err != nil {
 		return err
 	}
-	ctx = cecontext.WithTopic(ctx, topic)
-	return m.client.Send(ctx, event)
+
+	dt := extensions.FromSpanContext(trace.FromContext(ctx).SpanContext())
+	msg := new(pubsub.Message)
+	if err := cepubsub.WritePubSubMessage(ctx, binding.ToMessage(&event), msg, dt.WriteTransformer()); err != nil {
```

Comment:
It seems like we need to work with `message` instead of `event` for tracing support? In `handler.go`, we transform http request to message and then to event. Can we add the span context there to avoid another transformation?

Reply:
This is a bit of refactoring that will be needed before the event conversion can be elided as the SDK requires that all attribute and extension access be done through transformers.

Reply:
I will do this refactoring in a separate PR